### PR TITLE
Fixed certificates are not deleted on `make clean`

### PR DESCRIPTION
**Pull request by loewenstein**
Commit 224a9e25 recently made istio certs fixed by removing them from
.gitignore:
- security/docker/istio_ca.crt
- security/docker/istio_ca.key
- security/docker/node_agent.crt
- security/docker/node_agent.key

Hence `make clean` should no longer remove these.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -461,8 +461,7 @@ mixer/docker/ca-certificates.tgz security/docker/ca-certificates.tgz: docker/ca-
 FILES_TO_CLEAN+=mixer/docker/ca-certificates.tgz security/docker/ca-certificates.tgz
 
 # NOTE: this list is passed to rm -f during "make clean"
-GENERATED_CERT_FILES:=security/docker/istio_ca.crt security/docker/istio_ca.key \
-                      security/docker/node_agent.crt security/docker/node_agent.key
+GENERATED_CERT_FILES:=
```
[the remark this answers is not in the document]
Comment (mattdelco) [reply Makefile]: The more appropriate change is probably to just delete the "FILES_TO_CLEAN+=$(GENERATED_CERT_FILES)" line (a few lines below this one).  "GENERATED_CERT_FILES" is still needed, but used in a different way, in a pending PR that I've got.

File: security/Makefile [hunk 2]
```diff
@@ -23,8 +23,6 @@ check.prereqs:
 	../bin/verify_go_version.sh
 
 clean:
-	rm -f docker/istio_ca
```
Comment (mattdelco) [inline security/Makefile]: I'm not clear on the motivation for this.  These are deleting the go executables that get copied over from go/bin as part of making a docker image and are unrelated to the .crt/.key files.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (myidpt) [reply Makefile]: Why don't you delete "FILES_TO_CLEAN+=$(GENERATED_CERT_FILES)" line?

[the remark this answers is not in the document]
Comment (mangchiandjjoe) [reply Makefile]: We don't need to generate certificates and keys anymore in a build time.
security/bin/gen-keys.sh needs to be removed and Makefile should be updated.

@costinm Could you please take a look?

File: security/Makefile (hunk 2 above)
Comment (loewenstein) [reply security/Makefile]: You are right. Will remove this from the PR.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (loewenstein) [reply Makefile]: Alright, I just had these files in the index after `make clean` and tried to do a minimal change.

I will first update the PR removing `FILES_TO_CLEAN+=$(GENERATED_CERT_FILES)`, which is the minimal correct change to get rid of the file deletion in the index.

Then I will look into removing security/bin/gen-keys.sh altogether.

**Comment by loewenstein** [comment]
@mangchiandjjoe I removed security/bin/gen-keys.sh and the references in Makefile. Not sure how to "completely" verify nothing brakes... I did run the targets I touched, though.

`make docker.prebuilt` errs out on master as well for me.
```
cp /Users/d043856/workspace/go/bin/{pilot-test-client,pilot-test-server,pilot-test-eurekamirror} pilot/docker
cp: /Users/d043856/workspace/go/bin/pilot-test-client: No such file or directory
cp: /Users/d043856/workspace/go/bin/pilot-test-server: No such file or directory
cp: /Users/d043856/workspace/go/bin/pilot-test-eurekamirror: No such file or directory
make: *** [docker.prebuilt] Error 1
```

**Comment by mattdelco** [comment]
I don't understand the motivation of the earlier change to static files.  There might be some runtime overhead but otherwise if things are properly in the Makefiles and tests then people won't notice any othe consequences. The 100-year certs are unlikely to last that long given the pace at which standard/expectations change on encryption algorithms and key lengths, and whomever re-gens the certs might appreciate the file to refer to (granted it's still in the repo history but who's going to remember where to look for it).

In my larger pending PR it builds docker images out of a temp image, and as part of that invoked a slightly modified gen-keys.sh that can output to an alternate directory.  Eventually I had in mind that someone would just "port" gen-keys.sh into the makefile.  So, overall I don't like the direction that the prior PR and this one is headed.

As for this PR, I don't understand why the docker.istio-ca-test target is being deleted.  The security/docker/Dockerfile.istio-ca-test still exists, so either the target should still exist or the dockerfile should be deleted.

Overall I'd rather see the prior PR reverted and stat to require that people put explanatory comments in their PR.  Otherwise, for this PR I think the most that should be done is remove the:

FILES_TO_CLEAN+=$(GENERATED_CERT_FILES)

and remove the "security/bin/gen-keys.sh" from the recipe for "docker.prebuilt".

**Comment by loewenstein** [comment]
@costinm @mattdelco @mangchiandjjoe As this PR is apparently more controversial than I thought I took a step back, opened issue #2996 and created a new PR #2998 that only removes the certs from FILES_TO_CLEAN and thereby only fixes the immediate problem with `make clean`.

I suppose we can close this one.

**Comment by andraxylia** [comment]
Do we want this PR or not? If yes, please resolve conflicts.

**Comment by mattdelco** [comment]
I'll close this one out.  There's still the question about the tradeoffs between static vs generated cert files but I suspect that the author of this PR doesn't have a stake in the issue and was mostly motivated to address the symptoms of "make clean" causing changes/damage to the git source tree. If anyone wants to revisit the issue on static vs. generated certs then that is probably best handled in another PR, issue, and/or group discussion (and if no one is motivated to do any of these I may eventually create a PR to delete the now-unused static files from source).

**Pull request closed without merging**